eng/common/scripts/TypeSpec-Project-Generate.ps1

```diff
@@ -14,6 +14,81 @@ $ErrorActionPreference = "Stop"
 . $PSScriptRoot/common.ps1
 Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
 
+function GetNpmPackageVersion([string]$packagePath) {
+    $packagePath = Resolve-Path $packagePath
+    $packageFolder = Split-Path -Parent $packagePath
+    $packageName = Split-Path -Leaf $packagePath
+    Push-Location $packageFolder
+    try {
+        $packageVersion = node -p -e "require('./$packageName').version"
+        return $packageVersion
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+function GetEmitterDependencyVersion([string]$packagePath) {
+    $matchString = "`"$emitterName`": `"(.*?)`""
+    If ((Get-Content -Raw $packagePath) -match $matchString) {
+        return $Matches[1]
+    }
+}
+
+function NpmInstallAtRoot() {
+    $root = Resolve-Path "$PSScriptRoot/../../.."
+    Push-Location $root
+    try {
+        #default to root/eng/emitter-package.json but you can override by writing
+        #Get-${Language}-EmitterPackageJsonPath in your Language-Settings.ps1
+        $replacementPackageJson = "$PSScriptRoot/../../emitter-package.json"
+        if (Test-Path "Function:$GetEmitterPackageJsonPathFn") {
+            $replacementPackageJson = &$GetEmitterPackageJsonPathFn
+        }
+
+        $replacementVersion = GetEmitterDependencyVersion $replacementPackageJson
+        Write-Host "Replacement emitter version: $replacementVersion"
+
+        $installedPath = Join-Path $root "node_modules" $emitterName "package.json"
+        Write-Host "Checking installed emitter at $installedPath"
+        if (Test-Path $installedPath) {
+            $installedVersion = GetNpmPackageVersion $installedPath
+            Write-Host "Installed emitter version: $installedVersion"
+
+            if ($installedVersion -eq $replacementVersion) {
+                Write-Host "Emitter already installed. Skip installing."
+                return
+            }
+        }
+
+        Write-Host "Installing package at"(Get-Location)
+
+        if (Test-Path "node_modules") {
+            Write-Host "Remove existing node_modules at"(Get-Location)
+            Remove-Item -Path "node_modules" -Force -Recurse
+        }
+
+        $hasPackageFile = Test-Path "package.json"
+
+        $npmInstallCommand = "npm install --prefix $root $emitterName@$replacementVersion --no-package-lock --omit=dev"
```

Comment:
We should make an explicit decision to make this change as it has larger implications on other things like other dependencies. I don't want to change to this unless we have thought it through and switched to that strategy everywhere.

Reply:
@weshaggard , it's not aligned for all language emitters. I would suggest to use package.json for installation rather than install emitter individually. we could drop package.json to `sdk` folder to avoid conflicts with package.json in root folder.

Reply:
@weshaggard This is the case the extra dependency is NOT needed for TS emitter but required for TypeSpec files. 

Not only for package `@typespec/versioning`, but also for below packages if TypeSpec project referred as dependencies.

```
@azure-tools/typespec-autorest
@typespec/openapi
@typespec/openapi2
...
```
Adding these dependencies into emitter is okay but may introduce [phantom dependency issue](https://rushjs.io/pages/advanced/phantom_deps/). Not sure this is best practice or not.

Reply:
I tend to agree we should stick with a package.json file and I believe we can put it under the tempTypeSpec directory it just needs to live under the directory structure of wherever we put the TypeSpec project.

Reply:
@weshaggard , I think @pshao25 's purpose is to avoid multiple times installation for emitter. Put package.json under tempTypeSpec directory doesn't work and it should be put under upper parent folder like `sdk` folder.

Reply:
@weshaggard the design was that we didn't have any additional dependencies in the emitter-package.json.

Reply:
I don't think this will accommodate the devDependency pinning and emitter-package-lock.json work we're doing. We would prefer to `npm ci` using `emitter-package.json` and `emitter-package-lock.json` rather than explicit version installation.